bacon/model/StorageCouchBaseDB.swift

```diff
@@ -160,6 +165,40 @@ class StorageCouchBaseDB {
         }
     }
 
+    func deleteTransaction(_ transaction: Transaction) throws {
+        // Fetch the specific document from database
+        guard let transactionId = transactionMapping[transaction] else {
+            throw StorageError(message: """
+                Unable to find mapping of transaction object to it's unique id in the database.
+            """)
+        }
+        guard let transactionDocument = transactionDatabase.document(withID: transactionId) else {
+            throw StorageError(message: """
+                Unable to retrieve transaction document in database using id from mapping.
+            """)
+        }
+        log.info("""
+            StorageCouchBaseDB.deleteTransaction() with argument:
+            transaction:\(transaction).
+        """)
+        // Delete the document
+        do {
+            try transactionDatabase.deleteDocument(transactionDocument)
```

Comment:
Actually, what do you think about setting up observer-observable here, so that the dictionary is automatically kept up to date when transactionDatabase changes? Is it possible/feasible/worth the overhead?

Reply:
Kept up to date, meaning when creating, loading and deletion?
As of now, when transactions are created we don't keep a reference to it, this is because when the user wants to view this newly created transaction, he/she has to go to the transaction history page, which will require loading. 
So dictionary is only updated when transactions are loaded and deleted. (which is how it is implemented now)
I don't really see a need for it? I am open to be persuaded otherwise haha.

Reply:
Haha in that case I think this is fine, it was just a thought :P